### ISSUE TITLE
Installation issue: ELPA

### ISSUE THREAD

**Issue by mohan002** [LLNL/spack#20789]
Error:
==> Installing elpa-2017.05.001-fqahf335eay23rb35ibxdkf7y3gd6g2l
==> No binary for elpa-2017.05.001-fqahf335eay23rb35ibxdkf7y3gd6g2l found: installing from source
==> Fetching http://elpa.mpcdf.mpg.de/html/Releases/2017.05.001/elpa-2017.05.001.tar.gz
################################################################################### 100.0%
==> Warning: The contents of the archive look like HTML.  Either the URL you are trying to use does not exist or you have an internet gateway issue.  You can remove the bad archive using 'spack clean <package>', then try again using the correct URL.

gzip: stdin: not in gzip format
/usr/bin/tar: Child returned status 1
/usr/bin/tar: Error is not recoverable: exiting now
==> Error: ProcessError: Command exited with status 2:
    '/usr/bin/tar' '-oxf' '/tmp/mohbabul/spack-stage/spack-stage-elpa-2017.05.001-fqahf335eay23rb35ibxdkf7y3gd6g2l/elpa-2017.05.001.tar.gz'

### Steps to reproduce the issue

```console
$ spack install -v elpa@2017.05.001 +openmp %aocc@2.2.0 ^amdblis ^amdscalapack ^openmpi@4.0.3+cxx fabrics=auto
```

```
$ spack debug report
```
* **Spack:** 0.16.0-534-60a58d89e
* **Python:** 3.6.8
* **Platform:** linux-rhel8-zen2
* **Concretizer:** original

**Comment by scheibelp** [comment]
The http:// link that Spack is trying doesn't appear to work but the site does appear to respond to the https:// link. It looks like the `url_for_version` function of the ELPA package needs to be updated: 

```
    def url_for_version(self, version):
        t = 'https://elpa.mpcdf.mpg.de/html/Releases/{0}/elpa-{0}.tar.gz'  # <--- this is the only change: add an "s" to http
        if version < Version('2016.05.003'):
            t = 'http://elpa.mpcdf.mpg.de/elpa-{0}.tar.gz'
        return t.format(str(version))
```

With that do

```
spack clean --stage
spack install ... (or whatever you were doing)
```

Can you try that?

**Comment by mohan002** [comment]
@scheibelp that worked. thanks.

**Comment by adamjstewart** [comment]
@mohan002 want to submit a PR to fix this?

**Comment by mohan002** [comment]
Yeah sure @adamjstewart

**Comment by RemiLacroix-IDRIS** [comment]
Sorry I missed this issue but this problem is fixed in 5ef2daaa4a4a0b12c8adcedf496f6b9e61e8950e.
